examples/text_embedder/ios/TextEmbedder/ViewController.swift

```diff
@@ -0,0 +1,161 @@
+  @IBOutlet weak var clear2Button: UIButton!
+  @IBOutlet weak var compareButton: UIButton!
+  @IBOutlet weak var input1TextView: UITextView!
+  @IBOutlet weak var input2TextView: UITextView!
+
+  @IBOutlet weak var bottomSpaceLayoutConstraint: NSLayoutConstraint!
+  @IBOutlet weak var contentScrollView: UIScrollView!
+  @IBOutlet weak var resultLabel: UILabel!
+
+  let backgroundQueue = DispatchQueue(
+    label: "com.google.mediapipe.backgroundQueue",
+    qos: .userInteractive
+  )
+
+  private var textEmbedderService: TextEmbedderService?
+  private let textViewBotomSpace: CGFloat = 100
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    setupUI()
+    hideKeyboardOnBackgroundTap()
+    input1TextView.delegate = self
+    input2TextView.delegate = self
+
+    // Keyboard notification
+    NotificationCenter.default.addObserver(
+      self,
+      selector: #selector(keyboardWillChangeFrame), name: UIResponder.keyboardWillChangeFrameNotification,
+      object: nil)
+
+    // Setup the TextEmbedder Service.
+    backgroundQueue.async {
+      self.textEmbedderService = TextEmbedderService(modelPath: Constants.modelPath)
+    }
+  }
+
+  deinit {
+    NotificationCenter.default.removeObserver(self)
+  }
+
+  @IBAction func clearButton1TouchUpInside(_ sender: Any) {
+    input1TextView.text = ""
+    updateButtonState()
+  }
+  @IBAction func clearButton2TouchUpInside(_ sender: Any) {
+    input2TextView.text = ""
+    updateButtonState()
+  }
+
+  @IBAction func compareButtonTouchUpInside(_ sender: Any) {
+    guard let textEmbedderService = textEmbedderService else { return }
+    let text1 = input1TextView.text
+    let text2 = input2TextView.text
+    backgroundQueue.async { [weak self] in
+      guard let weakSelf = self else { return }
+      guard let similarity = textEmbedderService.compare(
+        text1: text1,
+        text2: text2) else {
+        DispatchQueue.main.async {
+          weakSelf.resultLabel.text = Constants.errorString
+        }
+        return
+      }
+      DispatchQueue.main.async {
+        weakSelf.resultLabel.text = String(format: "Similarity: %.2f", similarity)
+      }
+    }
+  }
+
+  // Private function
+  private func setupUI() {
+    input1TextView.text = Constants.defaultText
+    input2TextView.text = Constants.defaultText
+    input1TextView.inputAccessoryView = createInputAccessoryView()
+    input2TextView.inputAccessoryView = createInputAccessoryView()
+  }
+
+  private func hideKeyboardOnBackgroundTap() {
+    let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
+    tap.cancelsTouchesInView = false
+    view.addGestureRecognizer(tap)
+  }
+
+  private func createInputAccessoryView() -> UIView {
+    let inputAccessoryView = UIView(frame: CGRect(x: 0, y: 0, width: view.bounds.width, height: 44))
+    inputAccessoryView.backgroundColor = .lightGray
+
+    let doneButton = UIButton(frame: CGRect(x: view.bounds.width - 80, y: 5, width: 60, height: 30))
```

Comment:
add constants for these values. Like we have in the vision samples. width, spacing from left etc